src/Utility/ConstraintStabiliser.php

```diff
@@ -15,6 +15,27 @@ class ConstraintStabiliser
     private const GRAPHQL_3_VERSION_DEV = '3.8.x-dev';
     private const GRAPHQL_3_VERSION_STABLE = '~3.8.1@stable';
 
+    private static function getAllItems(LibraryRelease $libraryRelease)
+    {
+        $names = [];
+        $items = [];
+        $rootLibraryRelease = $libraryRelease;
+        // using a for loop instead of while loop to prevent infinite while loop
+        // if we change some code the wrong way in the future
+        for ($i = 0; $i < 10; $i++) {
+            $rootLibraryRelease = $rootLibraryRelease->getParentRelease() ?: $rootLibraryRelease;
+        }
```

Comment:
What does this do, exactly? Why are we specifying to do this 10 times specifically? Is the intention to just get the root (most-parent) release? If so, this is not a clean way to do that. Instead, implement a recursive `getRootRelease()` (or however you want to name it) method:
```php
public function getRootRelease(): ?LibraryRelease
{
    $parent = $this->getParentRelease();
    if ($parent) {
        return $parent->getRootRelease();
    }
    return $this;
}
```

Reply:
Updated

Reply:
Please implement `getRootRelease()` in the `LibraryRelease` class as that will make it easier to reuse if we need it in the future.